walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,81 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long duration = request.getDuration();
+
+    Collection<String> mandatory = request.getAttendees();
+
+
+    Collection<String> allAttendees = new ArrayList<>();
+    allAttendees.addAll(mandatory);
+    allAttendees.addAll(request.getOptionalAttendees());
+
+    Collection<TimeRange> availabilityAll = query(events, allAttendees, duration);
+    if (!availabilityAll.isEmpty() || mandatory.isEmpty()) {
+      return availabilityAll;
+    }
+
+    // Just return times for mandatory attendees.
+    return query(events, mandatory, duration);
```

Comment:
nit: can remove this, less spacing between.
```suggestion
    return query(events, mandatory, duration);
```

Reply:
Done